app/pages/patientdata/patientdata.js

```diff
@@ -590,18 +592,22 @@ export let PatientData = React.createClass({
 
   componentWillReceiveProps: function(nextProps) {
     var userId = this.props.currentPatientInViewId;
-    var currentPatientData = _.get(this.props, ['patientDataMap', userId], null);
-
     var nextPatientData = _.get(nextProps, ['patientDataMap', userId], null);
 
-    if (!currentPatientData && nextPatientData) {
+    // Hold processing until patient is fetched (ensuring settings are accessible), AND
+    // processing hasn't already taken place (this should be cleared already when switching patients), AND
+    // nextProps patient data exists
+    if (!nextProps.fetchingPatient && !this.state.processedPatientData && nextPatientData) {
       this.doProcessing(nextProps);
     }
   },
 
   doProcessing: function(nextProps) {
     var userId = this.props.currentPatientInViewId;
     var patientData = _.get(nextProps, ['patientDataMap', userId], null);
+    var patientSettings = _.get(nextProps, ['patient', 'settings'], null);
+    _.defaultsDeep(patientSettings, DEFAULT_SETTINGS);
```

Comment:
Just pushed that, so give it a shot. In the meantime, I'll keep trying to reproduce on my side. Really curious what specific state might be causing that.